### PR TITLE
With useTransition hook

### PR DESCRIPTION
**Pull request by rppld**
# Description

This PR adds a custom `useTransition()` hook to the library and replaces `react-spring` in `<Dialog />` and `<MenuButton />`.

## How to test

- Checkout this branch
- `$ yarn dev`
- Visit http://localhost:9001/?path=/story/dialog--basic
- Verify the mount/unmount transition works as expected
- Visit http://localhost:9001/?path=/story/menubutton--default
- Verify the mount/unmount transition works as expected

### CODE REVIEW

File: src/hooks/useTransition.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      }
+      let id = setTimeout(tick, duration)
+      return () => clearTimeout(id)
+    }
+
+    if (on && !prevOn) {
+      setTransitioning(true)
+      setShow(true)
+      requestAnimationFrame(() => {
+        requestAnimationFrame(handleMount)
```
Comment (RobVermeer) [inline src/hooks/useTransition.js:44]: Can you explain the need for 2 x `requestAnimationFrame`?

Comment (rppld) [reply src/hooks/useTransition.js:44]: https://gist.github.com/getify/3002042#gistcomment-2351663

If you’d only call it once, then the initial transition after initial page-load won’t work. It’s bit weird, but you can try it out by removing one call, refresh the page, and open a dialog. Alternatively a `setTimeout()` call with `0` delay also works, but I thought it was uglier than the above.

**Comment by RobVermeer** [comment]
Is it doable to remove all the `react-spring` stuff with this? :)

**Review by RobVermeer — APPROVED** [review]
Works as expected, API looks nice! 😄

**Pull request merged**